pkg/util/apiserver/cors_test.go

```diff
@@ -0,0 +1,141 @@
+		origin               string
+		expectedAllowOrigin  string
+		expectedAllowMethods []string
+		expectedAllowHeaders []string
+		args                 args
+	}{
+		{"With Verrazzano URI set", "https://console.somesuffix.xip.io",
+			"https://console.somesuffix.xip.io",
+			allowedHTTPMethods, allowedHeaders,
+			args{verrazzanoURI: "somesuffix.xip.io", writer: http.ResponseWriter(httptest.NewRecorder())}},
+
+		{"Without Verrazzano URI set", "", "",
+			allowedHTTPMethods, allowedHeaders,
+			args{verrazzanoURI: "", writer: http.ResponseWriter(httptest.NewRecorder())}},
+
+		{"Origin does not match", "http://localhost", "",
+			allowedHTTPMethods, allowedHeaders,
+			args{verrazzanoURI: "", writer: http.ResponseWriter(httptest.NewRecorder())}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			//GIVEN the Verrazzano URI suffix is set
+			instance.SetVerrazzanoURI(tt.args.verrazzanoURI)
+			req, err := http.NewRequest("GET", "http://someUrl", nil)
+			assert.Nil(t, err, "Error creating test request")
+			req.Header.Add("Origin", tt.origin)
+
+			//WHEN SetupOptionsResponse is called
+			SetupOptionsResponse(&tt.args.writer, req)
+
+			//THEN it writes the expected CORS header values
+			assert.Equal(t, tt.expectedAllowOrigin, tt.args.writer.Header().Get("Access-Control-Allow-Origin"))
+			assert.Equal(t, strings.Join(tt.expectedAllowMethods, ", "), tt.args.writer.Header().Get("Access-Control-Allow-Methods"))
+			assert.Equal(t, strings.Join(tt.expectedAllowHeaders, ", "), tt.args.writer.Header().Get("Access-Control-Allow-Headers"))
+		})
+	}
+}
+
+func Test_getAllowedOrigin(t *testing.T) {
+	vzURI := "something.v8o.oracledx.com"
+	defaultExpectedValue := fmt.Sprintf("https://console.%s", vzURI)
+	tests := []struct {
+		name              string
+		origin            string
+		additionalOrigins []string
+		originAllowed     bool
+	}{
+		{"origin matches default", defaultExpectedValue, []string{}, true},
+		{"origin matches default with additional origin", defaultExpectedValue, []string{"http://localhost:8000"}, true},
+		{"origin matches single additional origin", "http://localhost:8000", []string{"http://localhost:8000"}, true},
+		{"origin matches one of additional origins", "http://localhost:8888", []string{"http://localhost:8000", "http://localhost:8888"}, true},
+		{"origin matches default with multiple additional origins", defaultExpectedValue, []string{"http://localhost:8000", "http://localhost:8888"}, true},
+		{"negative test origin matches none", "http://somethingelse", []string{"http://localhost:8000", "http://localhost:8888"}, false},
+		{"negative test origin does not match default", "http://somethingelse", []string{}, false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// GIVEN the verrazzano URI suffix and an ACCESS_CONTROL_ALLOW_ORIGINS override env var
+			instance.SetVerrazzanoURI(vzURI)
+			os.Setenv("ACCESS_CONTROL_ALLOW_ORIGINS", strings.Join(tt.additionalOrigins, ", "))
```

Comment:
huh, do we need to set an environment variable? that seems brittle. tests can run in parallel too...

Reply:
yeah, I was a bit unsure about this when I did it too....wasn't sure if tests can run in parallel and if they share the same OS environment in that case. Let me see if there is an alternative

Reply:
I've updated to not set env vars in the tests and instead have the GetenvFunc overridable in env.go